pkg/k8s/apis/cilium.io/utils/repository.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package utils
+
+import (
+	slim_metav1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/policy/types"
+)
+
+// RulestoPolicyEntries converts the external rule type to the internal policy entry representation.
+func RulesToPolicyEntries(rules api.Rules) types.PolicyEntries {
+	entries := types.PolicyEntries{}
+	for _, rule := range rules {
+		es, node := getSelector(rule)
+
+		for _, iRule := range rule.Ingress {
+			defaultDeny := rule.EnableDefaultDeny.Ingress != nil && *rule.EnableDefaultDeny.Ingress
+			req := convertToLabelSelectorRequirementSlice(iRule.FromRequires)
+			l3 := types.ToEndpointSelectorInterfaceSlice(iRule.GetSourceEndpointSelectorsWithRequirements(req))
+			l3 = append(l3, types.ToEndpointSelectorInterfaceSlice(iRule.FromCIDR)...)
+			l3 = append(l3, types.ToEndpointSelectorInterfaceSlice(iRule.FromCIDRSet)...)
+			l4 := append(iRule.ToPorts, icmpRules(iRule.ICMPs)...)
```

Comment:
I forget the Go slice semantics here. Is there a chance we accidentally mutate `iRule`?